### PR TITLE
Ryan branch PR

### PR DESCRIPTION
**Pull request by rslaughter12**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,5 +14,11 @@
   "license": "ISC",
   "devDependencies": {
     "concurrently": "^5.3.0"
+  },
```
Comment (UserOlena) [inline package.json:17]: All this dependencies are related to the Client folder and shouldn't be installed in the root. I think your Pages and Components work because this packages are already installed in the Client package.json. Not sure that React can read this packages from the root node_modules folder. React reads node_modules from the SRC folder

File: client/src/components/Carousel.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
+import React, { useState, useEffect } from 'react';
```
Comment (UserOlena) [inline client/src/components/Carousel.js]: To follow the convention to keep our repo neat and organized, please include React in the { } along with the useState and useEffect like in the other files that we already have. 
Please go over all the files to resolve this issue to avoid same comments repetition with same problem in the other.

File: client/src/components/Carousel.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+import { makeStyles, useTheme } from '@material-ui/core/styles';
```
Comment (UserOlena) [inline client/src/components/Carousel.js:2]: Formatting convention that we are using is to structure components from the same package if there are few of them, like 
```
import {
  Avatar,
  Button,
} from '@mui/material';
``` 
You can look into the other files that we already have for a referral.

File: client/src/components/Carousel.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
```
Comment (UserOlena) [inline client/src/components/Carousel.js]: Formatting convention that we are using is to destructure components from the same package if there are few of them, like this if there are more then 3 of them and in line if there are only 2. We discussed it in the slack
```
import {
  Avatar,
  Button,
  TextField,
} from '@mui/material';
``` 
You can look into the other files that we already have for a referral.

Please go over all the files to resolve this issue to avoid same comments repetition with same problem in the other.

File: package.json [hunk 5]
```diff
@@ -14,5 +14,11 @@
...
+  "dependencies": {
+    "@emotion/react": "^11.11.1",
+    "@emotion/styled": "^11.11.0",
+    "@material-ui/core": "^4.12.4",
+    "@material-ui/icons": "^4.11.3"
   }
```
Comment (UserOlena) [inline package.json:23]: All this dependencies are related to the Client folder and shouldn't be installed in the root. I think your Pages and Components work because this packages are already installed in the Client package.json. Not sure that React can read this packages from the root node_modules folder. React reads node_modules from the SRC folder

File: package.json [hunk 6]
```diff
@@ -14,5 +14,11 @@
   "license": "ISC",
   "devDependencies": {
     "concurrently": "^5.3.0"
+  },
+  "dependencies": {
+    "@emotion/react": "^11.11.1",
+    "@emotion/styled": "^11.11.0",
+    "@material-ui/core": "^4.12.4",
+    "@material-ui/icons": "^4.11.3"
```
Comment (UserOlena) [inline package.json]: All this dependencies are related to the Client folder and shouldn't be installed in the root. I think your Pages and Components work because this packages are already installed in the Client package.json. Not sure that React can read this packages from the root node_modules folder. React reads node_modules from the SRC folder

File: client/src/components/Carousel.js [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
+
```
Comment (UserOlena) [inline client/src/components/Carousel.js:11]: Same here, formatting style that we are using

File: client/src/components/Carousel.js [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+  },
+  arrowButton: {
+    margin: theme.spacing(0, 1),
+    padding: theme.spacing(1),
+    color: theme.palette.type === 'dark' ? 'white' : 'black',
+  },
+}));
+
+const Carousel = () => {
+  const classes = useStyles();
```
Comment (UserOlena) [inline client/src/components/Carousel.js:47]: We have decided to use function declaration and arrow function only for the callback function. We had a long discussion in the slack before we came to this decision

File: client/src/components/Carousel.js [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+  const theme = useTheme();
+
+  const items = [
+    {
+      title: 'Item 1',
+      description: 'Description for Item 1',
```
Comment (UserOlena) [inline client/src/components/Carousel.js:53]: Do we need this array of objects? Our data will be coming from the DB

File: client/src/components/Carousel.js [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+      image: 'item3.jpg',
+    },
+    {
+      title: 'Item 4',
+      description: 'Description for Item 4',
+      image: 'item4.jpg',
+    },
+  ];
+  const [activeIndex, setActiveIndex] = useState(0);
+
```
Comment (UserOlena) [inline client/src/components/Carousel.js:73]: Should we keep all the useState variables at the top of the function for the styling convention and to easier find and maintenance them?

File: client/src/components/Categories.js [hunk 11]
```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+import { Button, Grid, Typography, Card, CardContent, Box, useTheme } from '@material-ui/core';
```
Comment (UserOlena) [inline client/src/components/Categories.js]: Same here as per line 1 and 2

File: client/src/components/Categories.js [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+
+// Sample data from the database
+const options = [
+  { name: 'Option 1', category: 'Category 1' },
+  { name: 'Option 2', category: 'Category 2' },
```
Comment (UserOlena) [inline client/src/components/Categories.js:15]: DB data will be used

File: client/src/components/Categories.js [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+  { name: 'Option 3', category: 'Category 1' },
+  // ... add more options
+];
+
+const Categories = () => {
+  const [selectedCategory, setSelectedCategory] = useState('Category 1');
```
Comment (UserOlena) [inline client/src/components/Categories.js:21]: Function style convenience

File: client/src/components/Categories.js [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+  const theme = useTheme();
+
+  // Get all unique categories from the options
+  const categories = [...new Set(options.map(option => option.category))];
+
```
Comment (UserOlena) [inline client/src/components/Categories.js:26]: If we will not be using an array with fake data, will we still need this variable?

File: client/src/components/Categories.js [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+      <Grid container spacing={2}>
+        <Grid item xs={12}>
+          <div className="category-buttons">
+            {categories.map(category => (
+              <Button
+                key={category}
+                variant="outlined"
+                size="small"
+                className={`category-button ${selectedCategory === category ? 'active' : ''}`}
+                onClick={() => selectCategory(category)}
```
Comment (UserOlena) [inline client/src/components/Categories.js:59]: Please format all the files with the Prettier extension to keep the same style convention within the repository and avoid shifted lines.

Please go over all the files to resolve this issue to avoid same comments repetition with same problem in the other.

**Comment by UserOlena** [comment]
After resolving merge conflict the app doesn't work. Probably the export issue and empty components/index.js are the reason

Screenshot 2023-07-16 at 10 38 45 PM

**Comment by UserOlena** [comment]
Usually, we provide a description for the PR for the reviewer to understand what to expect from the update and what to check. Please, go over previous commits to see the example.

Here is an [example](https://github.com/UserOlena/Learnify/pull/51)

File: client/src/components/index.js
Comment (sendusyourbones) [inline client/src/components/index.js:1]: What is this file being used for?

File: client/src/components/Carousel.js (hunk 2 above)
Comment (rslaughter12) [reply client/src/components/Carousel.js]: updated

File: client/src/components/Footer.js [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+      <div style={{ display: 'flex', alignItems: 'center' }}>
+        {/* Add your company logo here */}
+        <img src="/path/to/company-logo.png" alt="Company Logo" style={{ width: '50px', marginRight: '10px' }} />
+        <p>&copy; {currentYear} Your Company Name. All rights reserved.</p>
+      </div>
+      <div style={{ marginTop: '10px' }}>
+        <a href="/about" style={{ color: 'white', marginRight: '10px' }}>About</a>
+        <a href="/careers" style={{ color: 'white', marginRight: '10px' }}>Careers</a>
+        <a href="/who-we-are" style={{ color: 'white' }}>Who We Are</a>
+      </div>
```
Comment (UserOlena) [inline client/src/components/Footer.js:29]: href attribute links seems to be not related to out app

File: client/src/components/Carousel.js (hunk 9 above)
Comment (rslaughter12) [reply client/src/components/Carousel.js:53]: this is just a placeholder until the data is coming through so that I could see what then cards looked like on the page.

File: client/src/components/Categories.js (hunk 11 above)
Comment (rslaughter12) [reply client/src/components/Categories.js]: updated

File: client/src/components/Categories.js (hunk 12 above)
Comment (rslaughter12) [reply client/src/components/Categories.js:15]: this is just a placeholder until the data is coming through so that I could see what then cards looked like on the page.

File: client/src/components/index.js
Comment (rslaughter12) [reply client/src/components/index.js:1]: was deleted

File: client/src/components/Footer.js (hunk 16 above)
Comment (rslaughter12) [reply client/src/components/Footer.js:29]: those will be future projects to add about our buisness pages, about the program, and a basic careers page, was just a place holder option if we wanted it in the future.

**Comment by UserOlena** [comment]
I noticed that you are using px measure unit frequently that is not very convenient for different screen sizes. If user is to shrink their browser all the layout will shift in unexpected way. Instead fluid measure units are preferable such as em, rem, % and fluid formula for the font size that needs to be adjusted based on the styling goal

font-size: calc(14px + (24px - 14px) * ((100vw - 320px) / (1200px - 320px)));

**Comment by rslaughter12** [comment]
> I noticed that you are using px measure unit frequently that is not very convenient for different screen sizes. If user is to shrink their browser all the layout will shift in unexpected way. Instead fluid measure units are preferable such as em, rem, % and fluid formula for the font size that needs to be adjusted based on the styling goal
> 
> font-size: calc(14px + (24px - 14px) * ((100vw - 320px) / (1200px - 320px)));

File: client/src/components/Carousel.js (hunk 4 above)
Comment (rslaughter12) [reply client/src/components/Carousel.js]: updated

File: client/src/components/Carousel.js (hunk 7 above)
Comment (rslaughter12) [reply client/src/components/Carousel.js:11]: updated

File: client/src/components/Categories.js (hunk 13 above)
Comment (rslaughter12) [reply client/src/components/Categories.js:21]: updated

File: client/src/components/Carousel.js [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+    }, 5000);
+
+    return () => {
+      clearInterval(timer);
+    };
+  }, [items.length]);
+
+  const handlePrev = () => {
+    setActiveIndex((prevIndex) => {
+      if (prevIndex === 0) {
```
Comment (UserOlena) [inline client/src/components/Carousel.js]: Still a lot of arrow functions that are not callback functions in all the files

**Comment by UserOlena** [comment]
Ryan, you also modified SignIn and SignUp pages even so it wasn't part of this commit. It is a very bad practice to modify many files at once, because there is always a chance that something can go wrong and we will need to revert a merge commit. If this happens. we will lose a lot of code at once. It is more likely to happen after there was a merge conflict. It is better to separate this commit into smaller pieces especially since there were a lot of problems and updates. You can do `git add client/src/pages/Home.js` and any other related files like Home.css for the home page. Same for the other components. Smaller commits are more secure in case we need to revert as well as easier to review and to test. 

As per modified SignIn and SignUp...you shouldn't have modified them as part of this commit and you also used Prettier extension without configuring it to the agreement that we made in slack. I also slacked screenshot of configuration fot it after the team decided with styling preferences. Please re-read our chat to understand our styling convention that we all decided to use for our project, because it is very time consuming to go over this styling issues again and again as well as for you to redo it again and again rather than just read our requirements once and use that knowledge for the future updates.

File: client/src/components/Carousel.js (hunk 17 above)
Comment (rslaughter12) [reply client/src/components/Carousel.js]: adjusted

File: client/src/components/Categories.js (hunk 14 above)
Comment (sendusyourbones) [reply client/src/components/Categories.js:26]: If we're leaving the options placeholder array for now, we'll leave this for now.

File: client/src/components/Carousel.js (hunk 3 above)
Comment (sarah-jensen) [inline client/src/components/Carousel.js]: check package.json in client folder --> the import for material-ui v5 is '@mui/material' instead of '@material-ui/core'
so for all of the components/pages you'll want to check that:
'@material-ui/core' is changed to: '@mui/material' (anything that has core/other can just be grouped with @mui/material)
'@material-ui/icons-material' is changed to: '@mui/icons-material'
--I had to update imports in Carousel.js, Categories.js, and Navbar.js to resolve errors

I don't know what it replaces yet (and it's not an issue for this PR), but there's also '@mui/styled-engine' :)

Comment (rslaughter12) [inline client/src/components/Carousel.js]: is this resolved with the groups decision to go with version 4?

File: client/src/components/Carousel.js [hunk 18]
```diff
@@ -0,0 +1,140 @@
+import React, { useState, useEffect } from 'react';
+import { makeStyles, useTheme } from '@material-ui/core/styles';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
```
Comment (sarah-jensen) [inline client/src/components/Carousel.js]: these icon imports (lines 7 & 8) can be combined like this: 
`import { ArrowBack, ArrowForward } from "@material-ui/icons";`

the names of the components on lines 134 and 141 should have "Icon" removed from them to match the imports

File: client/src/components/Categories.js [hunk 19]
```diff
@@ -0,0 +1,101 @@
...
+  // Get all unique categories from the options
+  const categories = [...new Set(options.map(option => option.category))];
+
+  // Filter options based on selected category
+  const filteredOptions = selectedCategory
+    ? options.filter(option => option.category === selectedCategory)
+    : options;
+
+  // Function to handle category selection
+  const selectCategory = (category) => {
```
Comment (sarah-jensen) [inline client/src/components/Categories.js]: I believe this should be a function declaration instead of arrow

File: client/src/App.js [hunk 20]
```diff
@@ -1,7 +1,8 @@
 import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
 import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client';
 import './style/App.css';
-import { Home } from './pages';
+import Home from './pages/Home';
```
Comment (sarah-jensen) [inline client/src/App.js]: if you pull origin main, there's an import object for all of the pages that Home can be added to on line 4

File: client/src/components/Carousel.js (hunk 3 above)
Comment (sarah-jensen) [inline client/src/components/Carousel.js]: Since we're going to stick with v4, all of the imports (except the icons) can be in this object on line 2: 
`import { 
Card, 
CardContent, 
makeStyles, 
Typography, 
useTheme, 
IconButton 
} from "@material-ui/core";`

File: client/src/components/Categories.js [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+    setSelectedCategory(category);
+  };
+
+  return (
+    <Box
+      border={3}
+      borderRadius={8}
+      borderColor={theme.palette.mode === 'dark' ? 'black' : 'white'}
```
Comment (sarah-jensen) [inline client/src/components/Categories.js]: I believe this should be a function declaration instead of arrow

File: client/src/components/Categories.js [hunk 22]
```diff
@@ -0,0 +1,101 @@
...
+const Categories = () => {
+  const [selectedCategory, setSelectedCategory] = useState('Category 1');
+  const theme = useTheme();
+
+  // Get all unique categories from the options
+  const categories = [...new Set(options.map(option => option.category))];
+
+  // Filter options based on selected category
+  const filteredOptions = selectedCategory
+    ? options.filter(option => option.category === selectedCategory)
```
Comment (sarah-jensen) [inline client/src/components/Categories.js]: I believe this should be a function declaration instead of arrow

File: client/src/components/Categories.js [hunk 23]
```diff
@@ -0,0 +1,101 @@
...
+  { name: 'Option 3', category: 'Category 1' },
+  // ... add more options
+];
+
+const Categories = () => {
+  const [selectedCategory, setSelectedCategory] = useState('Category 1');
+  const theme = useTheme();
+
+  // Get all unique categories from the options
+  const categories = [...new Set(options.map(option => option.category))];
```
Comment (sarah-jensen) [inline client/src/components/Categories.js]: I believe this should be a function declaration instead of arrow

File: client/src/components/Navbar.js [hunk 24]
```diff
@@ -0,0 +1,159 @@
+import React, { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
```
Comment (sarah-jensen) [inline client/src/components/Navbar.js]: { makeStyles } can be included with the other import object from '@material-ui/core'

File: client/src/components/Navbar.js [hunk 25]
```diff
@@ -0,0 +1,159 @@
...
+  Toolbar,
+  Typography,
+  MenuItem,
+  Menu,
+  InputBase,
+  Button,
+  Switch,
+} from '@material-ui/core';
+import MenuIcon from '@material-ui/icons/Menu';
+import SearchIcon from '@material-ui/icons/Search';
```
Comment (sarah-jensen) [inline client/src/components/Navbar.js:12]: make icon imports into an object to reduce lines of code:
`import { Menu, Search } from '@material-ui/icons'`

File: client/src/components/Navbar.js [hunk 26]
```diff
@@ -0,0 +1,159 @@
...
+          variant="contained"
+          color="inherit"
+          className={classes.signUpButton}
+          onClick={handleMenuOpen}
+        >
+          Categories
+        </Button>
+        <div className={classes.search}>
+          <div className={classes.searchIcon}>
+            <SearchIcon />
```
Comment (sarah-jensen) [inline client/src/components/Navbar.js:112]: update name of icon component to `<Search />` to match import

File: client/src/components/Recommended.js [hunk 27]
```diff
@@ -0,0 +1,147 @@
+import React, { useState } from 'react';
+import { makeStyles, useTheme } from '@material-ui/core/styles';
```
Comment (sarah-jensen) [inline client/src/components/Recommended.js]: import object from `@material-ui/core` can include `makeStyles` and `useTheme`

check that you're importing the components from `@material-ui/core` (v4)

File: client/src/components/Recommended.js [hunk 28]
```diff
@@ -0,0 +1,147 @@
...
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
```
Comment (sarah-jensen) [inline client/src/components/Recommended.js]: combine icon imports into a single import object in case more icons are needed:
`import { ArrowBack, ArrowForward } from '@material-ui/icons'`

update component/icon names on lines 132 and 140

File: client/src/components/Recommended.js [hunk 29]
```diff
@@ -0,0 +1,147 @@
...
+import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
+
+const useStyles = makeStyles((theme) => ({
```
Comment (sarah-jensen) [inline client/src/components/Recommended.js]: change to function declaration? I don't think this is considered callback...

File: client/src/components/Recommended.js [hunk 30]
```diff
@@ -0,0 +1,147 @@
...
+    setCurrentIndex((prevIndex) => Math.min(prevIndex + 4, items.length - 4));
+  };
+
+  const visibleItems = items.slice(currentIndex, currentIndex + 4);
+
+  return (
+    <div className={classes.recommendations}>
+      <h2>Recommendations Component</h2>
+      <div className={classes.recommendationsContent}>
+        {visibleItems.map((item) => (
```
Comment (sarah-jensen) [inline client/src/components/Recommended.js:114]: I think this should be a function declaration instead of arrow

File: client/src/components/Recommended.js [hunk 31]
```diff
@@ -0,0 +1,147 @@
...
+        ))}
+      </div>
+      <div className={classes.recommendationsArrows}>
+        <IconButton
+          className={classes.arrowButton}
+          color="primary"
+          disabled={currentIndex === 0}
+          onClick={handlePrev}
+        >
+          <ArrowBackIcon />
```
Comment (sarah-jensen) [inline client/src/components/Recommended.js]: <ArrowBack /> to match import object

File: client/src/components/Recommended.js [hunk 32]
```diff
@@ -0,0 +1,147 @@
...
+        </IconButton>
+        <IconButton
+          className={classes.arrowButton}
+          color="primary"
+          disabled={currentIndex >= items.length - 4}
+          onClick={handleNext}
+        >
+          <ArrowForwardIcon />
```
Comment (sarah-jensen) [inline client/src/components/Recommended.js]: <ArrowForward /> to match import object

File: client/src/pages/Home.js [hunk 33]
```diff
@@ -1,10 +1,50 @@
...
-        <div>Homepage</div>
-    );
-}
+import React, { useState } from 'react';
+import { ThemeProvider, createMuiTheme } from '@material-ui/core/styles';
+import Navbar from '../components/Navbar';
+import Carousel from '../components/Carousel';
+import Recommendations from '../components/Recommended';
+import Categories from '../components/Categories';
+import Footer from '../components/Footer';
```
Comment (sarah-jensen) [inline client/src/pages/Home.js]: combine lines 3-7 into single import object:
`import { Carousel, Categories, Footer, Navbar, Recommendations } from '../components'`

File: package.json [hunk 34]
```diff
@@ -14,5 +14,11 @@
   "license": "ISC",
   "devDependencies": {
     "concurrently": "^5.3.0"
+  },
+  "dependencies": {
+    "@emotion/react": "^11.11.1",
```
Comment (sarah-jensen) [inline package.json]: if we're using v4 of material-ui, you can `npm uninstall @emotion/react @emotion/styled`

File: client/src/pages/Home.js [hunk 35]
```diff
@@ -1,10 +1,50 @@
...
+
+const Home = () => {
```
Comment (sarah-jensen) [inline client/src/pages/Home.js]: change to function declaration for the export to work, this will allow "Home" to be imported with the other pages in App.js rather than on a separate line
`export function Home() {`

File: client/src/pages/Home.js [hunk 36]
```diff
@@ -1,10 +1,50 @@
-import { Link } from 'react-router-dom';
-import { useQuery } from '@apollo/client';
-
-export const Home = () => {
-    return (
-        <div>Homepage</div>
-    );
-}
+import React, { useState } from 'react';
```
Comment (sarah-jensen) [inline client/src/pages/Home.js]: `useState` isn't used, maybe remove it from the import object?

**Review by sarah-jensen — COMMENTED** [review]
I realize this is an early commit, so I tried to only pick out comments based on what I saw in VS code after pulling what I believe is the most up to date version of the branch this evening. I left comments on these lines (the line numbers may differ in VS code, these are from Github):
App.js - line 4
Home.js - lines 1, 7, 9
Carousel.js - lines 2, 7, 8, 134, 141
Categories.js - lines 2, 25
Navbar.js - lines 2, 14, 112
Recommended.js - lines 2, 7, 10, 132, 140

I looked through your commits individually, and I think some of the arrow functions you updated in that commit got reverted --> I *think* I captured them in the lines above. 

There are also a good number of lingering double quotes, so double check that you updated your Prettier settings and do a quick reformat of your component and page files. :)

File: client/src/components/Carousel.js (hunk 3 above)
Comment (sarah-jensen) [reply client/src/components/Carousel.js]: they can be included in the import object with the rest of `'@material-ui/core'` instead of on a separate line, but otherwise yes :)

File: package.json [hunk 37]
```diff
@@ -16,6 +16,10 @@
     "concurrently": "^5.3.0"
   },
   "dependencies": {
+    "@emotion/react": "^11.11.1",
```
Comment (UserOlena) [inline package.json:19]: All this packages belong to the client folder. They shouldn't be installed in the root package.json

Comment (UserOlena) [inline package.json]: All the react packages belong to the client folder. They shouldn't be installed in the root package.json

File: client/src/components/index.js
Comment (UserOlena) [reply client/src/components/index.js:1]: This file was meant for importing of all the components so we could import those components by destructuring and specifying the path only 1 time rather then repeating the path for each component like this 

Screenshot 2023-07-18 at 9 49 26 AM

in order to follow the DRY principle of software development:

DRY "Don't repeat yourself", which stands for 'don't repeat yourself,' is a principle of software development that aims at reducing the repetition of patterns and code duplication in favor of abstractions and avoiding redundancy.

File: client/src/App.js [hunk 38]
```diff
@@ -1,11 +1,8 @@
 import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
 import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client';
 import './style/App.css';
-import { 
-  Home, 
-  SignUp, 
-  SignIn,
-} from './pages';
+import { SignUp, SignIn } from "./pages";
+import Home from "./pages/Home";
```
Comment (UserOlena) [inline client/src/App.js]: Home should be imported along with all other pages. If it doesn't work for you, export Home like all other pages are exported. I just pulled your branch, modified Home export like other pages and it works being imported on the one line with all other pages

File: client/src/components/Carousel.js (hunk 7 above)
Comment (UserOlena) [reply client/src/components/Carousel.js:11]: The rule that we follow "If we have more than 2 components or attributes for an element, we break each one of them into the new line.

**Comment by UserOlena** [comment]
When I open your branch on a bigger screen, layout seems fine. When I move it to my 14" screen size laptop and that is not the smallest laptop size, things start to shift, like buttons text don't really fit in the background

Screenshot 2023-07-18 at 9 45 51 AM

and footer elements overlap

Screenshot 2023-07-18 at 10 10 17 AM

I suggested earlier for you to change measure unit from `px` to `em`, `rem`, `%` and `fluid formula` for the font size to avoid such behavior, While I understand it can be adjusted later using media queries, but there are so many different screen sizes, and it is impossible and time consuming to create a break point for each one of them, while using fluid measure units mostly solve this problem, as it adapts to the screen view port size. It is easier to just use this measure units from the very beginning styling process to avoid re-doing all of that for different break points.
But it can be done at the later stage

File: client/src/App.js (hunk 38 above)
Comment (rslaughter12) [reply client/src/App.js]: As I said earlier, for some reason it is not working, I believe I have exported it correctly, doing this fixes it on my end. I do not think that this one line being slightly out of format is going to impact anything. It works, I am going for minimal viable product right now and this gets me there. We can adjust it on future versions if it turns out to be a game changer.

**Comment by rslaughter12** [comment]
> 

I am going for minimal viable product. Getting elements on the page, then making them functional with the backend server. After the site is functional is when the majority of the styling will take place. I understand your preference of doing things but I also would like the way that I style to be respected as well. This push's purpose is to get the page in the main branch so that it can be connected to the server.

File: client/src/components/Carousel.js (hunk 18 above)
Comment (rslaughter12) [reply client/src/components/Carousel.js]: It is currently this version
import { React, useState, useEffect } from 'react';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import { Card, CardContent, Typography, IconButton } from '@mui/material';
import ArrowBackIcon from '@mui/icons-material/ArrowBack';
import ArrowForwardIcon from '@mui/icons-material/ArrowForward';

the arrows break when put together like that, this keeps them functional

File: client/src/components/Categories.js (hunk 19 above)
Comment (rslaughter12) [reply client/src/components/Categories.js]: updated
  function selectCategory(category) {
    setSelectedCategory(category);
  }

File: client/src/App.js (hunk 20 above)
Comment (rslaughter12) [reply client/src/App.js]: For some reason ( im 90% sure all is right in my system) this does not work. Keeping it separate works. I think it should stay this way.

File: client/src/components/Carousel.js (hunk 3 above)
Comment (rslaughter12) [reply client/src/components/Carousel.js]: updated import { React, useState, useEffect } from 'react';
import {  Card,  CardContent,  makeStyles,  Typography,  useTheme,  IconButton  } from "@material-ui/core";
import ArrowBackIcon from '@mui/icons-material/ArrowBack';
import ArrowForwardIcon from '@mui/icons-material/ArrowForward';

File: client/src/components/Navbar.js (hunk 24 above)
Comment (rslaughter12) [reply client/src/components/Navbar.js]: updated
import { React, useState } from 'react';
import {
  AppBar,
  Toolbar,
  Typography,
  MenuItem,
  Menu,
  InputBase,
  Button,
  makeStyles
} from '@material-ui/core';
import SearchIcon from '@material-ui/icons/Search';

File: client/src/components/Navbar.js (hunk 25 above)
Comment (rslaughter12) [reply client/src/components/Navbar.js:12]: this was updated in a later commit, 
import { React, useState } from 'react';
import {
  AppBar,
  Toolbar,
  Typography,
  MenuItem,
  Menu,
  InputBase,
  Button,
  makeStyles
} from '@material-ui/core';
import SearchIcon from '@material-ui/icons/Search';

Comment (rslaughter12) [reply client/src/components/Navbar.js:12]: menu icon was switched out for a button

File: client/src/components/Navbar.js (hunk 26 above)
Comment (rslaughter12) [reply client/src/components/Navbar.js:112]: the name of the import is SearchIcon as only one is being imported from icons.
format will change if more icons are added to the file.

File: client/src/components/Recommended.js (hunk 27 above)
Comment (rslaughter12) [reply client/src/components/Recommended.js]: adjusted
import { React, useState } from 'react';
import { makeStyles, useTheme } from '@material-ui/core';
import { Card, CardContent, Typography, IconButton } from '@mui/material';
import ArrowBackIcon from '@mui/icons-material/ArrowBack';
import ArrowForwardIcon from '@mui/icons-material/ArrowForward';

File: client/src/components/Recommended.js (hunk 28 above)
Comment (rslaughter12) [reply client/src/components/Recommended.js]: this was updated in a later commit, arrow icons are not working when called together, because of that they get their own line to make it functional.
import { React, useState } from 'react';
import { makeStyles, useTheme } from '@material-ui/core';
import { Card, CardContent, Typography, IconButton } from '@mui/material';
import ArrowBackIcon from '@mui/icons-material/ArrowBack';
import ArrowForwardIcon from '@mui/icons-material/ArrowForward';

File: client/src/components/Recommended.js (hunk 31 above)
Comment (rslaughter12) [reply client/src/components/Recommended.js]: this works with the way it is being called currently

File: client/src/components/Recommended.js (hunk 32 above)
Comment (rslaughter12) [reply client/src/components/Recommended.js]: this works with the way it is being called currently

File: client/src/components/index.js
Comment (rslaughter12) [reply client/src/components/index.js:1]: i fixed it

File: client/src/pages/Home.js (hunk 33 above)
Comment (rslaughter12) [reply client/src/pages/Home.js]: updated

File: package.json (hunk 34 above)
Comment (rslaughter12) [reply package.json]: updated

File: client/src/pages/Home.js (hunk 35 above)
Comment (rslaughter12) [reply client/src/pages/Home.js]: updated

File: client/src/pages/Home.js (hunk 36 above)
Comment (rslaughter12) [reply client/src/pages/Home.js]: updated

File: package.json (hunk 37 above)
Comment (rslaughter12) [reply package.json]: was uninstalled after Sarah's comment from yesterday,

File: client/src/components/Recommended.js (hunk 29 above)
Comment (rslaughter12) [reply client/src/components/Recommended.js]: that's the way that I believe useStyles was formatted in class, I think we should keep it that way.

Comment (sarah-jensen) [reply client/src/components/Recommended.js]: fair point! let's leave it then

File: client/src/components/Carousel.js (hunk 18 above)
Comment (sarah-jensen) [reply client/src/components/Carousel.js]: so the current imports for the arrow icons are version 5. since we agreed to use version 4, the imports should be updated for consistency's sake and so we can uninstall redundant packages.

I recommended combining the icon imports into one line for a couple of reasons: 1-the formatting/syntax is consistent with the other imports, and 2 - it's refactored in advance should we end up adding more icons later. If you feel strongly about it, that's fine, but I'd say if we add one more icon it should be refactored for consistency's sake.

what I discovered for both v4 and v5 is that for the imports to work in a single import object, the "Icon" suffix needs to be removed from the component name, and the name should also be removed from the path. I don't know why mui is weird that way, but it is!

File: client/src/App.js (hunk 20 above)
Comment (sarah-jensen) [reply client/src/App.js]: The function declaration and export was missing in the Home.js file, which is why it wasn't working. I see in the comments below that you fixed that, so this should work on your machine now, too! :)

File: client/src/components/Recommended.js (hunk 27 above)
Comment (sarah-jensen) [reply client/src/components/Recommended.js]: You'll want to update the Card, CardContent, Typography, IconButton components to be from version 4 as well (@material-ui.core)

basically, if anything is being imported from @mui____ it's v5 and should be switched over to v4

**Comment by sarah-jensen** [comment]
Hey Ryan!
I looked through the comments, and reopened a few (3 I think - one each from Carousel.js, App.js, and Recommended.js).

For the App.js one - since you updated the function component to be exported in Home.js the refactor to include Home with the other imports in App.js should work for you now.

The others are related to switching back to material ui v4 (it would probably be quickest to just double check that you're not importing anything from a package that starts with '@mui' since that's version 5 (I think all version 4 things start with '@material-ui/').

Something else I noticed with the imports was just having them formatted to be on separate lines (having 3+ on separate lines was something we had consensus on last night):
image

I'm sorry I didn't notice that earlier! If you want to send me a direct message after your next update I'll prioritize a review so we can get this meaty branch merged asap!

File: client/src/App.js (hunk 20 above)
Comment (rslaughter12) [reply client/src/App.js]: this has been updated in Ryan-Branch as well, I noticed I fixed it in another branch I was working in.

File: client/src/components/Carousel.js (hunk 18 above)
Comment (rslaughter12) [reply client/src/components/Carousel.js]: this has been fixed, will be committed shortly

File: client/src/components/Recommended.js (hunk 27 above)
Comment (rslaughter12) [reply client/src/components/Recommended.js]: updated will commit shortly

**Review by sarah-jensen — APPROVED** [review]
This is great Ryan! Any lingering formatting that we find can just be fixed as we move forward. :)
